src/core/server/integration_tests/saved_objects/migrations/kibana_migrator_test_kit.ts

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import { defaultsDeep } from 'lodash';
+import { BehaviorSubject, firstValueFrom, map } from 'rxjs';
+import { ConfigService, Env } from '@kbn/config';
+import { getEnvOptions } from '@kbn/config-mocks';
+import { getPackages } from '@kbn/repo-packages';
+import { REPO_ROOT } from '@kbn/repo-info';
+import { KibanaMigrator } from '@kbn/core-saved-objects-migration-server-internal';
+
+import {
+  SavedObjectConfig,
+  type SavedObjectsConfigType,
+  type SavedObjectsMigrationConfigType,
+  SavedObjectTypeRegistry,
+} from '@kbn/core-saved-objects-base-server-internal';
+import { SavedObjectsRepository } from '@kbn/core-saved-objects-api-server-internal';
+import {
+  ElasticsearchConfig,
+  type ElasticsearchConfigType,
+} from '@kbn/core-elasticsearch-server-internal';
+import { AgentManager, configureClient } from '@kbn/core-elasticsearch-client-server-internal';
+import { type LoggingConfigType, LoggingSystem } from '@kbn/core-logging-server-internal';
+import { docLinksServiceMock } from '@kbn/core-doc-links-server-mocks';
+
+import type { ISavedObjectTypeRegistry, SavedObjectsType } from '@kbn/core-saved-objects-server';
+import { esTestConfig, kibanaServerTestUser } from '@kbn/test';
+import { LoggerFactory } from '@kbn/logging';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { registerServiceConfig } from '@kbn/core-root-server-internal';
+
+export const defaultLogFilePath = Path.join(__dirname, 'kibana_migrator_test_kit.log');
+
+// Extract current stack version from Env, to use as a default
+const currentVersion = Env.createDefault(REPO_ROOT, getEnvOptions()).packageInfo.version;
+
+export interface KibanaMigratorTestKitParams {
```

Comment:
Can we also add explicit typing for the return value of `getKibanaMigratorTestKit`?

Reply:
Sure!